### PR TITLE
Allow non-semver packages in the Conan package registry

### LINKED ISSUE #20405: Package: Could not validate one or more reference fields
### Description

Using conan package manager with gitea 1.17.0 when uploading some of the packages gitea return this error:

```
❯ conan upload libbacktrace -r=gitea --force --confirm --all
Uploading to remote 'gitea':
Uploading libbacktrace/cci.20210118 to remote 'gitea'
ERROR: libbacktrace/cci.20210118: Upload recipe to 'gitea' failed: {"message":"Could not validate one or more reference fields"}
. [Remote: gitea]

ERROR: Errors uploading some packages
```

The log on gitea side is:

```
2022/07/19 12:01:58 ...vices/auth/oauth2.go:37:CheckOAuthAccessToken() [T] [62d68f26] oauth2.ParseToken: unexpected signing algo: HS256
...
2022/07/19 12:01:58 models/user/user.go:996:GetUserByIDCtx() [I] [62d68f26] [SQL] SELECT ...
...
2022/07/19 12:01:58 ...s/conan/reference.go:43:NewRecipeReference() [T] [62d68f26-3] Conan Recipe: libbacktrace/cci.20210118(@_/_(#))
2022/07/19 12:01:58 ...ges/helper/helper.go:32:LogAndProcessError() [D] [62d68f26-3] Could not validate one or more reference fields
2022/07/19 12:02:00 ...ource/manager_run.go:47:Run() [T] [62d68f1e-42] Event source has no listeners
2022/07/19 12:02:00 ...ource/manager_run.go:57:Run() [T] [62d68f1e-42] Pausing the eventsource
```

Is there a reason to why this package failed to be uploaded to gitea? most of them are uploaded with success this one no...it seems a name check?? openssl also fails for example:

```
❯ conan upload openssl -r=gitea --force --confirm --all
Uploading to remote 'gitea':
Uploading openssl/1.1.1q to remote 'gitea'
Uploading openssl/3.0.5 to remote 'gitea'
Uploaded conan_export.tgz -> openssl/3.0.5 [0.24k]
Uploaded conanfile.py -> openssl/3.0.5 [32.13k]
Uploaded conanmanifest.txt -> openssl/3.0.5 [0.10k]
Uploaded conan recipe 'openssl/3.0.5' to 'gitea': http://wolfserver:3000/api/packages/nuno/conan
Uploading package 1/1: 0a969fabc5bb3a3c9bdf93aca6ce2de02eb34d3a to 'gitea'
Uploaded conan_package.tgz -> openssl/3.0.5:0a96 [19184.23k]
Uploaded conaninfo.txt -> openssl/3.0.5:0a96 [2.49k]
Uploaded conanmanifest.txt -> openssl/3.0.5:0a96 [8.34k]
ERROR: openssl/1.1.1q: Upload recipe to 'gitea' failed: {"message":"Could not validate one or more reference fields"}
. [Remote: gitea]

ERROR: Errors uploading some packages
~ ❯  
```
full log
https://gist.github.com/nesc1/562b1e25d6132b4f091aa009b3f011ab

### Gitea Version

1.17.0+rc1-33-g780b198997

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/nesc1/562b1e25d6132b4f091aa009b3f011ab

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

dowloaded from your serves and running on a raspberry pi.

### Database

MySQL

### PR DESCRIPTION
**Pull request by algernon**
A lot of existing packages do not conform to SemVer, yet, they should be allowed in the Conan package registry as-is. To achieve this, remove the SemVer check from `NewRecipeReference`, and replace it with a simple empty string check.

A unit test with a non-semver version is also included.

Fixes #20405.

### CODE REVIEW

File: modules/packages/conan/reference.go [hunk 1]
```diff
@@ -56,7 +54,7 @@ func NewRecipeReference(name, version, user, channel, revision string) (*RecipeR
 	if !namePattern.MatchString(name) {
 		return nil, ErrValidation
 	}
-	if _, err := goversion.NewSemver(version); err != nil {
+	if (version == "") {
```
Comment (6543) [inline modules/packages/conan/reference.go]:
```suggestion
	if version == "" {
```

**Review by lafriks — APPROVED** [review]

File: modules/packages/conan/reference.go [hunk 2]
```diff
@@ -56,7 +54,7 @@ func NewRecipeReference(name, version, user, channel, revision string) (*RecipeR
 	if !namePattern.MatchString(name) {
 		return nil, ErrValidation
 	}
-	if _, err := goversion.NewSemver(version); err != nil {
+	if version == "" {
```
Comment (KN4CK3R) [inline modules/packages/conan/reference.go]:
```suggestion
	if strings.TrimSpace(version) == "" {
```
I would disallow whitespaces.

**Review by 6543 — APPROVED** [review]

File: modules/packages/conan/reference.go [hunk 3]
```diff
@@ -56,7 +54,7 @@ func NewRecipeReference(name, version, user, channel, revision string) (*RecipeR
 	if !namePattern.MatchString(name) {
 		return nil, ErrValidation
 	}
-	if _, err := goversion.NewSemver(version); err != nil {
+	if strings.TrimSpace(version) == "" {
```
Comment (6543) [inline modules/packages/conan/reference.go]: We should use trimed value

**Review by 6543 — CHANGES_REQUESTED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Comment by 6543** [comment]
please send backport

**Comment by 6543** [comment]
-> #20523

**Pull request merged**